sdk/storage/azure-storage-files-shares/src/share_file_client.cpp

```diff
@@ -1014,12 +1015,13 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
     auto uploadPageFunc = [&](int64_t offset, int64_t length, int64_t chunkId, int64_t numChunks) {
       (void)chunkId;
       (void)numChunks;
+      Storage::Details::FileReader fileReader(fileName);
```

Comment:
Because of how the newly added length validation and the overload that finds the file size on the behalf of the caller, we either need to open a new `FILE*` each time and forbid users from using existing handles while the `FileBodyStream` ctor is running, OR "use a non-portable method to get the file size that's less subject to races".

What do others think?

Thanks @CaseyCarter for the suggestions.

Reply:
No, it will exhaust file descriptors soon on Linux.

"use a non-portable method to get the file size that's less subject to races" is an option.